### PR TITLE
Okio upgrade post kotlin

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/internal/Util.java [hunk 1]
```diff
@@ -678,4 +679,20 @@ public static boolean sameConnection(HttpUrl a, HttpUrl b) {
...
+   *
+   * @param buffer the in-memory buffer
+   * @return byte returned by buffer.readByte()
+   * @throws AssertionError in place of EOFException
+   */
+  public static byte readIOSafeByte(Buffer buffer) {
+    try {
+      return buffer.readByte();
+    } catch (EOFException eofe) {
+      throw new AssertionError("unexpected eod of stream", eofe);
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/internal/Util.java]: eod -> eof

File: okhttp/src/main/java/okhttp3/internal/Util.java [hunk 2]
```diff
@@ -678,4 +679,20 @@ public static boolean sameConnection(HttpUrl a, HttpUrl b) {
...
+
+  /**
+   * Read a byte from a buffer where the operation is known to be safe, e.g. reading
+   * a stream in-memory with remaining bytes.
+   *
+   * @param buffer the in-memory buffer
+   * @return byte returned by buffer.readByte()
+   * @throws AssertionError in place of EOFException
+   */
+  public static byte readIOSafeByte(Buffer buffer) {
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/internal/Util.java]: I don't get the motivation for this method. Is it to avoid a checked exception?

Comment (yschimke) [reply okhttp/src/main/java/okhttp3/internal/Util.java]: Yep, an incompatibility between okio 1 and 2.  public method that now throws an IOException.

Comment (yschimke) [reply okhttp/src/main/java/okhttp3/internal/Util.java]: Do you want me to remove and add throws to each of the callers?

Comment (JakeWharton) [reply okhttp/src/main/java/okhttp3/internal/Util.java]: Huh, why wasn't this caught?

Comment (JakeWharton) [reply okhttp/src/main/java/okhttp3/internal/Util.java]: https://github.com/square/okio/commit/487a1f2392d8a9f017dceaca15656a82ab4ed46d

Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/Util.java]: Yeah, please add throws to the callers. I'm expecting no new catch clauses.

Comment (yschimke) [reply okhttp/src/main/java/okhttp3/internal/Util.java]: I had a branch&PR previously with the changes to switch the java code to okio 2.  

https://github.com/square/okhttp/pull/4256/files

**Pull request merged**